src/libLocalDescriptorCache/DescDbProxy.cpp

```diff
@@ -0,0 +1,319 @@
+    return m_localDB.FindFileByName(filename, output);                 
+}
+
+bool DescDbProxy::FindFileContainingSymbol(const std::string& symbol_name,
+                                        grpc::protobuf::FileDescriptorProto* output)
+{
+    return m_localDB.FindFileContainingSymbol(symbol_name, output);                 
+}                              
+
+bool DescDbProxy::FindFileContainingExtension(const std::string& containing_type,
+                                        int f_field_number,
+                                        grpc::protobuf::FileDescriptorProto* output)
+{
+    return m_localDB.FindFileContainingExtension(containing_type, f_field_number, output);                 
+}       
+
+bool DescDbProxy::GetServices(std::vector<grpc::string>* output)
+{
+    if(! m_serviceList.empty())
+    { 
+        (*output) = m_serviceList;
+        return true;
+    } 
+    else
+    {
+        std::cerr << "Error while fetching services from localDB entry for host "  << std::endl;
+        return false;    
+    }
+}  
+
+bool DescDbProxy::isValidHostEntry(const localDescDb::DescriptorDb& f_descDb, const std::string f_hostAddress)
+{    
+    bool validHostEntry=false; //TODO: Maybe rename in something with Reflection
+    static google::protobuf::util::TimeUtil timestamp;
+    google::protobuf::Timestamp currentTime;
+    google::protobuf::Timestamp lastUpdate;
+    int secCurrentTime;
+    int secLastUpdate;
+    int timeDiff;
+
+    for (int i=0; i < f_descDb.hosts_size(); i++)
+    {
+        const localDescDb::Host host = f_descDb.hosts(i);
+        if (host.hostaddress() == f_hostAddress)
+        {
+           // Get current Time and Time of last Upate in Seconds:
+            currentTime = timestamp.GetCurrentTime(); // time in UTC
+            secCurrentTime = timestamp.TimestampToSeconds(currentTime);
+            
+            lastUpdate = host.lastupdate();
+            secLastUpdate = timestamp.TimestampToSeconds(lastUpdate);
+
+            timeDiff = secCurrentTime - secLastUpdate;
+
+            // TODO: Change back to 120
+            // Expiry for validity of an entry
+            if (timeDiff <= 12)
+            {
+                validHostEntry = true;
+            }      
+        }
+    }
+
+    return validHostEntry;
+}
+
+// Remove all instances in DB od same host
+// Writes all other hosts in new messages and retuns this new message into descDb variable
+// Updates Db-file on disc.
+void deleteDuplicateHostEntries(localDescDb::DescriptorDb& descDb, const std::string f_hostAddress, const std::string f_dbFile)
+{
+    localDescDb::DescriptorDb newDescDb;
+    for(int i=0; i<descDb.hosts_size(); i++)
+    {
+        localDescDb::Host host = descDb.hosts(i);
+        if(host.hostaddress() != f_hostAddress)
+        {
+            localDescDb::Host* newHost = newDescDb.add_hosts(); //returns host
+            newHost->CopyFrom(host);
+        }
+    }
+
+    descDb = newDescDb;
+
+    // Overwrite DB-file on disc
```

Comment:
do not write back to HDD in this function. One function should do one thing: This function deletes duplicates from the RAM representation.
Then have another function to store Ram representation to HDD.
Otherwise one might have the idea later to call this in a loop without realizing HDD access is done here.